cmake/AwsCFlags.cmake

```diff
@@ -37,6 +37,8 @@ function(aws_set_common_properties target)
         list(APPEND AWS_C_FLAGS /volatile:iso)
         # Disable unknown pragma warnings
         list(APPEND AWS_C_FLAGS /wd4068)
+        # Disable non-constant aggregate initializer warnings
+        list(APPEND AWS_C_FLAGS /wd4204)
```

Comment:
IIRC the main reason we had this still on was to prevent it from creeping into headers and breaking C89 compat in our headers. Since we don't really put such code in headers except via macro, it's *probably* ok, but @JonathanHenson would be the most likely to recall some terrible reason for not doing this one.

Reply:
Yeah, don't put that one on the whole project, it should be disabled in msvc specific c files only or test files. Adding this will cause us to miss C++ breaking headerfile changes.

Reply:
Done.